tiledb/sm/misc/comparators.h

```diff
@@ -65,34 +61,34 @@ class RowCmp {
    * @param b The second coordinate.
    * @return `true` if `a` precedes `b` and `false` otherwise.
    */
-  bool operator()(const ResultCoords<T>& a, const ResultCoords<T>& b) const {
-    for (unsigned int i = 0; i < dim_num_; ++i) {
-      if (a.coords_[i] < b.coords_[i])
+  bool operator()(const ResultCoords& a, const ResultCoords& b) const {
+    for (unsigned int d = 0; d < dim_num_; ++d) {
+      auto res = domain_->cell_order_cmp(d, a.coord(d), b.coord(d));
+
+      if (res == -1)
         return true;
-      if (a.coords_[i] > b.coords_[i])
+      if (res == 1)
         return false;
-      // else a.coords_[i] == b.coords_[i] --> continue
+      // else same coordinate on dimension d --> continue
     }
 
     return false;
   }
 
  private:
+  /** The domain. */
+  const Domain* domain_;
   /** The number of dimensions. */
   unsigned dim_num_;
 };
 
 /** Wrapper of comparison function for sorting coords on col-major order. */
-template <class T>
 class ColCmp {
  public:
-  /**
-   * Constructor.
-   *
-   * @param dim_num The number of dimensions of the coords.
-   */
-  ColCmp(unsigned dim_num)
-      : dim_num_(dim_num) {
+  /** Constructor. */
+  ColCmp(const Domain* domain)
+      : domain_(domain) {
+    dim_num_ = domain->dim_num();
```

Comment:
same question here, i think we should stick with using list initialization wherever possible